causalml/propensity.py

```diff
@@ -47,6 +52,9 @@ def fit(self, X, y):
             X (numpy.ndarray): a feature matrix
             y (numpy.ndarray): a binary target vector
         """
+        if self.check_if_randomized(X, y):
```

Comment:
@yungmsh what are your thoughts on adding a boolean to the meta learner class itself that can bypass this all-together if user explicitly inputs something like `is_randomized_treatment` or `is_experimental` or something like that? It could possibly bypass a lot of compute time?

Reply:
Not a bad idea- we could add that so it bypasses it altogether, and even if user doesn't provide an argument for is_randomized_treatment (presumably defaulted to None),under the hood we can do a check anyway to prevent the repeated non-convergence situation

Reply:
@steveyang90 I thought about it again and think it's better to leave it to the user to provide a constant vector of propensities, and in the case the user doesn't provide it, it'll just do a single fit before detecting that it should be 0.5. Main reasons being i) additional code complexity doesn't justify minimal burden placed on the user - our code would have to account for both single treatment case (vector) and multiple treatment case (dict), and ii) leaving these learners as abstracted as possible is a positive thing. Happy to hear others thoughts here though.